layers/core_checks/cc_spirv.cpp

```diff
@@ -555,172 +579,255 @@ bool CoreChecks::ValidateCooperativeMatrix(const spirv::Module &module_state, co
         return ss.str();
     };
 
+    auto print_flexible_properties = [this]() {
+        std::ostringstream ss;
+        for (uint32_t i = 0; i < cooperative_matrix_flexible_dimensions_properties.size(); ++i) {
+            const auto &prop = cooperative_matrix_flexible_dimensions_properties[i];
+            ss << "[" << i << "] MGranularity = " << prop.MGranularity << " | NGranularity = " << prop.NGranularity
+               << " | KGranularity = " << prop.KGranularity << " | AType = " << string_VkComponentTypeKHR(prop.AType)
+               << " | BType = " << string_VkComponentTypeKHR(prop.BType) << " | CType = " << string_VkComponentTypeKHR(prop.CType)
+               << " | ResultType = " << string_VkComponentTypeKHR(prop.ResultType) << " | scope = " << string_VkScopeKHR(prop.scope)
+               << " | workgroupInvocations = " << prop.workgroupInvocations << '\n';
+        }
+        return ss.str();
+    };
+
     for (const spirv::Instruction *cooperative_matrix_inst : module_state.static_data_.cooperative_matrix_inst) {
         const spirv::Instruction &insn = *cooperative_matrix_inst;
         switch (insn.Opcode()) {
             case spv::OpTypeCooperativeMatrixKHR: {
-                CoopMatType m(insn.Word(1), module_state, stage_state);
+                CoopMatType m(insn.Word(1), module_state, stage_state, IsSignedIntType(insn.Word(2)));
+
+                if ((entrypoint.stage & VK_SHADER_STAGE_COMPUTE_BIT) != 0) {
+                    if (SafeModulo(local_size_x, phys_dev_props_core11.subgroupSize) != 0) {
+                        const auto vuid_string = m.scope == VK_SCOPE_SUBGROUP_KHR
+                                                     ? "VUID-VkPipelineShaderStageCreateInfo-module-08987"
+                                                     : "VUID-VkPipelineShaderStageCreateInfo-module-10169";
+                        skip |= LogError(vuid_string, module_state.handle(), loc,
+                                         "SPIR-V (compute stage) Local workgroup size in the X dimension (%" PRIu32
+                                         ") is not multiple of subgroupSize (%" PRIu32 ")/.",
+                                         local_size_x, phys_dev_props_core11.subgroupSize);
+                    }
+                    if (m.scope == VK_SCOPE_WORKGROUP_KHR) {
+                        const auto workgroup_size = local_size_x * local_size_y * local_size_z;
+                        if (workgroup_size >
+                            phys_dev_ext_props.cooperative_matrix_props2_nv.cooperativeMatrixWorkgroupScopeMaxWorkgroupSize) {
+                            skip |= LogError(
+                                "VUID-VkPipelineShaderStageCreateInfo-module-10169", module_state.handle(), loc,
+                                "SPIR-V (compute stage) Total local workgroup size (%" PRIu32
+                                ") is larger than cooperativeMatrixWorkgroupScopeMaxWorkgroupSize (%" PRIu32 ")/.",
```

Comment:
this seems like a typo

```suggestion
                                ") is larger than cooperativeMatrixWorkgroupScopeMaxWorkgroupSize (%" PRIu32 ").",
```

Reply:
done